src/runtime-rs/crates/runtimes/virt_container/src/container_manager/io/shim_io.rs

```diff
@@ -61,7 +73,10 @@ impl ShimIo {
                 .open(&stdin)
                 .await
             {
-                Ok(file) => Some(Box::new(file)),
+                Ok(file) => {
+                    set_flag_with_blocking(file.as_raw_fd());
```

Comment:
Since this is not intuitive, maybe add a comment like
```Rust
// Set it to blocking to avoid infinitely handling EAGAIN when the reader is empty
```

Reply:
Thanks, @Ji-Xinyou already added a comment.